balancer-js/src/modules/joins/joins.module.ts

```diff
@@ -155,6 +169,18 @@ export class Join {
       wrapMainTokens
     );
 
+    let totalBptZeroPi = BigNumber.from('0');
+    const leafTokens = PoolGraph.getLeafAddresses(orderedNodes);
+    leafTokens.forEach((leafToken) => {
+      const leafInputNodes = orderedNodes.filter(
+        (node) => node.address === leafToken
```

Comment:
I believe it's safer to make both lower case so the comparison is case insensitive.